CHANGELOG.md

```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Breaking changes
 
+* On MacOS, jj will now look for its config in `$XDG_CONFIG_HOME` and error if config is found in `~/Library/Application Support/jj/`
```

Comment:
I thought, when I originally saw this, that the idea was to *add* support for `$XDG_CONFIG_HOME` as a configurable default, but not *remove* support for the OS default. I am just one guy, but I *really* prefer things to go in the OS’ preferred locations instead of in `~/.config`. I have kind of resigned myself to having a bunch of junk in `~/.config`, but I am a Mac user first and foremost, and do not share a lot of the XDG philosophy or any commitment to it.

My own *very* strong preference here (and people who actually do work maintaining jj should of course just overrule this, but I wanted to flag it up) is to support both on macOS, given that I know lots of people *do* prefer the XDG location over `~/Library/Application Support`. There is a design question there about what the default should be, given a lot of users *do* use macOS as a nice UI and good hardware but fundamentally as a \*nix.

To make the point another way: imagine doing this to Windows users and telling them that `%APPDATA` was deprecated and they needed to just deal with having the XDG config added to `C:\Documents and Settings\chris\.config`. That… would go over badly. 😂 As it should!

Reply:
Fair, I'll keep Application Support as a fallback.